src/api/v1/prerequisites/is_sibling_user.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * "user_details-parent" route pre-requisite
+ *
+ * Determines if the authenticated user is trying to access an data
+ * of a sibling at the same organisation.
+ *
+ * Conditions under which this is true for user X attemptying to access user Y's data:
+ *
+ * - User X is an VOLUNTEER_ADMIN for a community business for which user Y is a VOLUNTEER
```

Comment:
Parent is only mentioned in the scope related to this prereq. `VOLUNTEER_ADMIN` needs to be able to access user details from other users at the same cb. Until this point:
- user details have been owned by the user
- user details have only been able to be accessed by `TWINE_ADMIN` and `ORG_ADMIN`
- `ORG_ADMIN` are considered owners of all the resources at their cb
- user hierarchy:
```
TWINE_ADMIN
|
|__ ORG_ADMIN (owns cbs 🏢 data)
  |
  |__ VOLUNTEER_ADMIN (owns own 👩🏻‍💼 data)
    |
    |__ VISITOR (owns own 👩🏼‍💼 data)
    |
    |__ VOLUNTEER (owns own 👩🏽‍💼 data)
```

`user_details-parent` is a new scope that allows `VOLUNTEER_ADMIN` to access user details at their org. I am treating`VOLUNTEER_ADMIN` as a higher level `VOLUNTEER`, rather than an `ADMIN`. This may not be the best name as the user details are owned by a (marginally) lower level user. `user_details-sibling` would be more ideal, but I'd prefer not to add extra permission levels unless they have other usage.

Reply:
After discussion, happy to have this renamed to `user_details-sibling`.

Reply:
Also, generalise the bullet point to more accurately describe the body of the pre-req (i.e. this is only restricting requests to be on the same organisation as the user is authenticated with).